one_fm/api/api.py

```
@@ -100,6 +100,17 @@ def rename_post(posts):
         except Exception as e:
             print(frappe.get_traceback())
 
+# This function allows you to fetch the list of Shift Permission of a given employee.
+# params: employee_ID (eg: HR-EMP-00001)
+# returns: List of shift Permission with name, date and workflow_state of the doc.
+@frappe.whitelist()
+def list_shift_permission(employee_id):
+    try:
+        shift_permission = frappe.get_list("Shift Permission", filters={'employee':employee_id}, fields=["name","date","workflow_state"])
+        return shift_permission
+    except Exception as e:
+        print(frappe.get_traceback())

```

Please return the error response instead of printing it.
Example: `return frappe.utils.response.report_error(e.http_status_code)`